Content.Server/Fluids/EntitySystems/AbsorbentSystem.Footprints.cs

```diff
@@ -0,0 +1,43 @@
+using System.Linq;
+using Content.Shared.Chemistry.Components;
+using Content.Shared.Fluids;
+using Content.Shared.FootPrint;
+using Content.Shared.Timing;
+
+namespace Content.Server.Fluids.EntitySystems;
+
+public sealed partial class AbsorbentSystem
+{
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+
+    /// <summary>
+    ///     Tries to clean a number of footprints in a range determined by the component. Returns the number of cleaned footprints.
+    /// </summary>
+    private int TryCleanNearbyFootprints(EntityUid user, EntityUid used, EntityUid target, AbsorbentComponent component,  Entity<SolutionComponent> absorbentSoln)
+    {
+        var footprintQuery = GetEntityQuery<FootPrintComponent>();
+        var targetCoords = Transform(target).Coordinates;
+        var entities = _lookup.GetEntitiesInRange(targetCoords, component.FootprintCleaningRange, LookupFlags.Uncontained);
+
+        // Take up to [MaxCleanedFootprints] footprints closest to the target
+        var cleaned = entities.AsEnumerable()
+            .Where(footprintQuery.HasComp)
+            .Select(uid => (uid, dst: Transform(uid).Coordinates.TryDistance(EntityManager, _transform, targetCoords, out var dst) ? dst : 0f))
+            .Where(ent => ent.dst > 0f)
+            .OrderBy(ent => ent.dst)
+            .Select(ent => (ent.uid, comp: footprintQuery.GetComponent(ent.uid)));
```

Comment:
kinda gross to be fair. Also no need for using `Where(footprintQuery.HasComp)` if you're going with the comment I left above.